### PR TITLE
UDF Cleanup

### PR DESCRIPTION
**Pull request by VamsiModem**

### CODE REVIEW

File: src/libraries/Microsoft.PowerFx.Core/Functions/UserDefinedFunction.cs [hunk 1]
```diff
@@ -71,7 +69,7 @@ public bool TryGetArgIndex(string argName, out int argIndex)
             return false;
         }
 
-        public TexlBinding BindBody(INameResolver nameResolver, IBinderGlue documentBinderGlue, BindingConfig bindingConfig = null, IUserDefinitionSemanticsHandler userDefinitionSemanticsHandler = null, Features features = null, INameResolver functionNameResolver = null, IExternalRule rule = null)
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Functions/UserDefinedFunction.cs:74]: >IUserDefinitionSemanticsHandler [](http://example.com/codeflow?start=132&length=31)

can we remove this definition completely?

File: src/libraries/Microsoft.PowerFx.Core/Syntax/UserDefinitions.cs [hunk 2]
```diff
@@ -106,12 +89,14 @@ private IEnumerable<UserDefinedFunction> CreateUserDefinedFunctions(IEnumerable<
                     continue;
                 }
 
-                if (!(CheckParameters(udf.Args, errors) & CheckReturnType(udf.ReturnType, errors)))
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Syntax/UserDefinitions.cs:109]: >&  [](http://example.com/codeflow?start=56&length=2)

For context:
[https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/boolean-logical-operators](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/boolean-logical-operators)

File: src/libraries/Microsoft.PowerFx.Core/Functions/UserDefinedFunction.cs (hunk 1 above)
Comment (VamsiModem) [reply src/libraries/Microsoft.PowerFx.Core/Functions/UserDefinedFunction.cs:74]: Done!

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by MikeStall — APPROVED** [review]

**Comment by LucGenetier** [comment]
✅ No public API change.

**Pull request merged**